lib/Sema/CSFix.cpp

```diff
@@ -2041,6 +2040,26 @@ bool AllowUnsupportedRuntimeCheckedCast::diagnose(const Solution &solution,
   return failure.diagnose(asNote);
 }
 
+AllowCheckedCastToUnrelated *
+AllowCheckedCastToUnrelated::attempt(ConstraintSystem &cs, Type fromType,
+                                     Type toType, CheckedCastKind kind,
+                                     ConstraintLocator *locator) {
+  // Explicit optional-to-optional casts always succeed because a nil
+  // value of any optional type can be cast to any other optional type.
+  if (fromType->getOptionalObjectType() && toType->getOptionalObjectType()) {
+    return nullptr;
```

Comment:
I was looking at the CSApply changes and it makes me think that we just return `optionalToOptionalCast` flag to `typeCheckCheckedCast` instead because although it wouldn't reproduce a fix here it would change the output of the `typeCheckCheckedCast` in CSApply where `Unresolved` is an actual error condition in some cases.

Reply:
Adjusted!